### DISCUSSION TITLE
Ability to specify timeline to start playback

### DISCUSSION THREAD

**Discussion by sshearer** [dialogic-godot/dialogic#645] [Ideas]
By pressing F5, I can start my 'video' and watch it playback according to the series of timelines I've created and linked together with the 'Change Timeline' control.  Works like a charm :)  However, now that I'm defining more content in each timeline it's taking longer & longer for me to get to the place where content has been created or changed in order for me to validate those changes.  I'd like to be able to select a specific timeline and press some kind of 'Play' button and have it start playback from there.

At the moment, I have just 1 'scene' and 1 Dialogic component/instance that has this series of timelines.  Is the solution for me to have 1 scene & Dialogic control *per timeline* and then use the 'Change Scene' control instead ?  If so, I'm not sure if Dialogic can be added many times into a project...

**Comment by Jowan-Spooner** [comment]
Not sure if I understand correcty. Can't you just change what timeline you start at in the DialogNode or the Dialogic.start() method? That would be what I'd do as long as I work on a scene. Then when I want I just change it back to the first timeline and can play the full game.

**Reply by sshearer** [reply]
;)  Suurrreeeee, provide the quick and easy solution ...  :)

I guess I was looking at the problem like a content creator, not a developer.  Thank you !

**Reply by abstractqqq** [reply]
Playback is not supported in Dialogic I believe. What do you mean playback exactly?  Like in timeline 1, you have dialog 1,2,3,4,...,100. You are currently on dialog 100, and you want to revisit dialog 99, and then 98, ...... ... ?

**Reply by sshearer** [reply]
The previous answer is probably the easiest solution at this point, but:

You sort-of have the right idea - more like I'd like to start at timeline19, step1.  Or eventually, start at timeline39,  step1.  I think you'd always need to start at step1 of any specific timeline because, in theory, each timeline should be independent & self-contained.   In my mind, I was equating it to standard IDE debugging features where you select a timeline and press 'play'.  I think it'd be difficult to start in the middle of a timeline because there's usually some amount of setup needed (eg. Which character should be active at the moment with what background image ?)

**Reply by abstractqqq** [reply]
You can take a look at my project (GodetteVNFramework), which has playback. But it won't playback events like characters moving (The animation won't be reversed because that is a little to complicated). It remembers all the background, music, character positions, and some screen effects. It allows you start at any index in a timeline, but if you start at an index, say 100, there will be no playback records for events before that because technically you never played events before that. But you can skip to the point where you want to start playing back.. Again, the playback is imperfect because it omits animations, and only remembers positions.
